src/main/java/edu/harvard/iq/dataverse/dataaccess/DataFileZipper.java

```diff
@@ -133,12 +138,26 @@ public long addFileToZipStream(DataFile dataFile, boolean getOriginal) throws IO
 
                 Success = false;
             } else {
+                String folderName = dataFile.getFileMetadata().getDirectoryLabel(); 
+                if (folderName != null) {
+                    while (folderName.startsWith("/")) {
```

Comment:
I'm a bit confused by this section. Does this preserve the nested folders? It seems like this strips all the folders until the last one.

A comment on what this section is doing could be useful for future maintenance.

Reply:
No, it only strips the leading slashes, if there are any. "///foo/bar" will become "foo/bar". 
I'll add a comment yes.

Reply:
Ah ok that makes sense! I was getting substring() mixed up with split()